pylearn2/datasets/dataset.py

```diff
@@ -102,6 +178,8 @@ def get_topo_batch_axis(self):
         Returns the index of the axis that corresponds to different examples
         in a batch when using topological_view.
         """
+        warnings.warn('This method is deprecated and will be unsupported '
+                      'after 27 July 27, 2015')
```

Comment:
I think these warnings should be in the docstring as well, since the `warn` would probably never be called.
It should still be part of DenseDesignMatrix, though, and I think the warning should reflect that, and say it is not going to be part of `Dataset`, but some `DenseDesignMatrix` would still implement it.